packages/flutter/lib/src/painting/text_style.dart

```diff
@@ -390,19 +432,39 @@ class TextStyle extends Diagnosticable {
       wordSpacing: wordSpacing == null ? null : wordSpacing * wordSpacingFactor + wordSpacingDelta,
       textBaseline: textBaseline,
       height: height == null ? null : height * heightFactor + heightDelta,
-      decoration: decoration,
-      decorationColor: decorationColor,
-      decorationStyle: decorationStyle,
+      decoration: decoration ?? this.decoration,
+      decorationColor: decorationColor ?? this.decorationColor,
+      decorationStyle: decorationStyle ?? this.decorationStyle,
+      debugLabel: modifiedDebugLabel,
     );
   }
 
-  /// Returns a new text style that matches this text style but with some values
-  /// replaced by the non-null parameters of the given text style. If the given
-  /// text style is null, simply returns this text style.
+  /// Returns a new text style that is a combination of this style and the given
+  /// [other] style, following the following rules.
+  ///
+  /// If the given [other] text style has its [TextStyle.inherit] set to true,
+  /// its null properties are replaced with the non-null properties of this text
+  /// style. It is said that the [other] style inherits properties of this
+  /// style. Another way to think of it is that the "missing" properties of the
+  /// [other] style are filled by the properties of this style.
+  ///
+  /// If the given [other] text style has its [TextStyle.inherit] set to false,
+  /// simply returns the given [other] style unchanged. It is said that the
+  /// [other] style does not inherit properties of this style.
+  ///
+  /// If the given text style is null, simply returns this text style.
```

Comment:
similarly with "simply". It just leads to people muttering "doesn't seem so simple to _me_" to themselves.

Reply:
not that i ever mutter to myself when reading documentation. no sir.